src/librustc_codegen_ironox/back/write.rs

```diff
@@ -25,8 +25,12 @@ fn write_module(
     module: &ModuleIronOx,
     path: &PathBuf,
     diag_handler: &Handler) {
-    if let Err(e) = fs::write(path, module.asm()) {
-        diag_handler.err(&format!("failed to write asm file {}", e));
+    // If this is a regular module, write it to a file. 'Metdata' and
```

Comment:
"Metdata" -> "Metadata"?

Reply:
And if we can't handle those, are we able to put in some sort of assert that will trigger if we do see those modules? Even if it's something like dumb like seeing if the filename contains a relevant substring?

Reply:
`codegen` is currently ignoring them, because I don't want compilation to fail. If `metadata`/`allocator` modules come first, `codegen` won't have a chance to compile the `regular` modules...

(I think there is a `metadata` module for every `regular` module, but I am not sure).

Reply:
Ah, I see. Maybe you can slightly clarify the comment for dummies like me that didn't get this?